common/metrics/tags.go

```diff
@@ -38,15 +38,24 @@ const (
 	// Generic reason tag can be used anywhere a reason is needed.
 	reason = "reason"
 	// See server.api.enums.v1.ReplicationTaskType
-	replicationTaskType     = "replicationTaskType"
-	replicationTaskPriority = "replicationTaskPriority"
-
-	namespaceAllValue = "all"
-	unknownValue      = "_unknown_"
-	totalMetricSuffix = "_total"
-	tagExcludedValue  = "_tag_excluded_"
-
-	errorPrefix = "*"
+	replicationTaskType                 = "replicationTaskType"
+	replicationTaskPriority             = "replicationTaskPriority"
+	versioningBehavior                  = "versioning_behavior"
+	isFirstAttempt                      = "first-attempt"
+	workflowStatus                      = "workflow_status"
+	workflowBehaviorBeforeOverride      = "workflow_behavior_before_override"
+	workflowBehaviorAfterOverride       = "workflow_behavior_after_override"
+	versioningOverrideOnNewWorkflow     = "versioning_override_on_new_workflow"
+	effectiveDeploymentBeforeTransition = "effective_deployment_before_transition"
+	effectiveDeploymentAfterTransition  = "effective_deployment_after_transition"
```

Comment:
let's use shorter names otherwise writing queries will be hard. some of the context is passed automatically by the _metric_ name itself.
```suggestion
	behaviorBefore      = "behavior_before"
	behaviorAfter       = "behavior_after"
	runInitiation     = "run_initiation" // see my other comment.
	fromUnversioned = "from_unversioned"
	toUnversioned  = "to_unversioned"
```